include/alpaka/mem/global/DeviceGlobalCpu.hpp

```diff
@@ -0,0 +1,68 @@
+/* Copyright 2024 Aurora Perego
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+#include "alpaka/mem/buf/cpu/Copy.hpp"
+#include "alpaka/mem/view/ViewPlainPtr.hpp"
+
+#include <type_traits>
+
+// memcpy specialization for device global variables
+namespace alpaka
+{
+    template<typename TViewSrc, typename TViewDstFwd, typename TQueue>
+    ALPAKA_FN_HOST auto memcpy(TQueue& queue, alpaka::DevGlobal<TViewDstFwd>& viewDst, TViewSrc const& viewSrc) -> void
+    {
+        typedef std::remove_all_extents_t<TViewDstFwd> T;
+        auto extent = getExtents(viewSrc);
+        auto view = alpaka::ViewPlainPtr<DevCpu, T, alpaka::Dim<decltype(extent)>, alpaka::Idx<decltype(extent)>>(
+            reinterpret_cast<T*>(&viewDst),
```

Comment:
IIUC, this relies on the fact that `DevGlobal<Type>` and `DevGlobal<Type>::value` are at the same address in memory.
If that is the case, please add a comment to explain it.

Reply:
(If I did it correctly) there is the overload of `operator&` that returns the address of `value`

Reply:
Mhm, ok. Why is the `reinterpret_cast<T*>` needed, then ?

Reply:
because `&viewDst` is a pointer to `TViewDstFwd` that can have an extent, while `T` is without the extent.
In the test for example `TViewDstFwd=unsigned int[3][2]`, `&viewDst` has type `unsigned int (*)[3][2]`, while `T=unsigned int`